### PR TITLE
CB-10324: Add entitlement for EFS in CloudBreak

### PR DESCRIPTION
**Pull request by LinaAtAustin**
Add entitlement CDP_DATA_LAKE_AWS_EFS in CloudBreak, so we can hide the EFS related code before it is fully functional.

This is blocked by https://github.infra.cloudera.com/thunderhead/thunderhead/pull/3031, which should be committed first, then commit this change.

### CODE REVIEW

File: mock-thunderhead/src/main/resources/application.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ auth:
     mediumdutysdx.enable: true
     upgrade.internalrepo.enable: true
     hbase.cloudstorage.enable: true
+    datalake.efs.enable: true
```
Comment (doktoric) [inline mock-thunderhead/src/main/resources/application.yml:32]: this means that locally we will always use EFS. This is what you want to achieve here ?

Comment (LinaAtAustin) [reply mock-thunderhead/src/main/resources/application.yml:32]: I can turn it off, and only enable in my local dev.

**Comment by bergerdenes** [comment]
Do you have a counterpart entitlement in the thunderhead as well?

**Comment by LinaAtAustin** [comment]
Yes. It is in https://jira.cloudera.com/browse/ALTCORE-3497, already committed. Its code review is at https://github.infra.cloudera.com/thunderhead/thunderhead/pull/3031

**Review by bergerdenes — COMMENTED** [review]
lgtm

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**